dhis-2/dhis-support/dhis-support-commons/src/test/java/org/hisp/dhis/commons/config/jackson/JacksonObjectMapperConfigTest.java

```diff
@@ -46,38 +50,54 @@
  */
 public class JacksonObjectMapperConfigTest
 {
-    private ObjectMapper jsonMapper = JacksonObjectMapperConfig.staticJsonMapper();
+    private final ObjectMapper jsonMapper = JacksonObjectMapperConfig.staticJsonMapper();
 
     @Test
     public void testIsoDateSupport()
-        throws JsonProcessingException
     {
-        Map<String, Date> yearTest = jsonMapper.readValue( createDateTest( "2019" ), new DateMapTypeReference() );
-        assertEquals( yearTest.get( "date" ), DateUtils.parseDate( "2019" ) );
-        Map<String, Date> yearMonthTest = jsonMapper.readValue( createDateTest( "2019-01" ),
-            new DateMapTypeReference() );
-        assertEquals( yearMonthTest.get( "date" ), DateUtils.parseDate( "2019-01" ) );
+        assertParsedAsDate( DateUtils.parseDate( "2019" ), "2019" );
+        assertParsedAsDate( DateUtils.parseDate( "2019-01" ), "2019-01" );
+        assertParsedAsDate( DateUtils.parseDate( "2019-01-01" ), "2019-01-01" );
+        assertParsedAsDate( DateUtils.parseDate( "2019-01-01T11:55" ), "2019-01-01T11:55" );
+        assertParsedAsDate( DateUtils.parseDate( "2019-01-01T11:55:01.444Z" ), "2019-01-01T11:55:01.444Z" );
+
     }
 
     @Test
-    public void testUnixEpochTimestamp()
-        throws JsonProcessingException
+    public void testIsoDateSupport_UnsupportedFormats()
     {
-        Map<String, Date> unixEpochDateString = jsonMapper.readValue( createDateTest( "1575118800000" ),
-            new DateMapTypeReference() );
-        assertEquals( unixEpochDateString.get( "date" ), new Date( 1575118800000L ) );
+        assertNotParsedAsDate( "2019-01-01T11:55:01.4444" );
```

Comment:
Note that DHIS2 ISO format does not allow for this which is unfortunate as this is the default output of `DateTimeFormatter.ISO_LOCAL_DATE_TIME.format`

Reply:
Created #7726 for it.